File: EntityFX.NetBenchamarks.Core/Generic/ArithmeticsBase.cs
using System.Runtime.CompilerServices;

namespace EntityFX.NetBenchmark.Core.Generic
{

    public abstract class ArithmeticsBase<TResult> : BenchmarkBase<TResult>
    {
        protected double R;

        public ArithmeticsBase()
        {
            Iterrations = 300000000;
            Ratio = 0.04;
        }

        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        protected static double DoArithmetics(long i)
        {
            return (i % 10) * (i % 100) * (i % 100) * (i % 100) * 1.11 + (i % 100) * (i % 1000) * (i % 1000) * 2.22 - i * (i % 10000) * 3.33 + i * 5.33;
        }
 
    }
}

File: src/python/entityfx/arithmetics_base.py

from entityfx.benchmark_base import BenchmarkBase
from entityfx.writer import Writer
import math

class ArithmeticsBase(BenchmarkBase):
    
    def __init__(self, writer : Writer=None, print_to_console : bool=True) -> None:
        super().__init__(writer, print_to_console)
        self._r = 0
        self._iterrations = 300000000
        self.ratio = .03
    
    @staticmethod
    def _doArithmetics(i : int) -> float:
        return ((((((math.floor(i / 10))) * ((math.floor(i / 100))) * ((math.floor(i / 100)))) * ((math.floor(i / 100))) * 1.11) + ((((math.floor(i / 100))) * ((math.floor(i / 1000))) * ((math.floor(i / 1000)))) * 2.22)) - ((i) * ((math.floor(i / 10000))) * 3.33)) + ((i) * 5.33)
    
    @staticmethod
    def _doArithmetics0(i : int) -> float:
        return ((((((math.floor(i / (10)))) * ((math.floor(i / (100)))) * ((math.floor(i / (100))))) * ((math.floor(i / (100)))) * 1.11) + ((((math.floor(i / (100)))) * ((math.floor(i / (1000)))) * ((math.floor(i / (1000))))) * 2.22)) - ((i) * ((math.floor(i / (10000)))) * 3.33)) + ((i) * 5.33)